update readme (#1)
DevExpress-Examples/xaf-how-to-use-the-file-attachment-module-with-a-legacy-database: Readme.md

<!-- default badges list -->
[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T237508)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

# XAF - How to use the File Attachment Module with a legacy database

> The [File Attachments](https://docs.devexpress.com/eXpressAppFramework/112781/document-management/file-attachments-module) module ships with the capability to easily implement a file upload and download functionality in your application. However, the built-in `FileData` class is persistent and therefore requires a separate table in the database to store file contents and names. These specifics make this class useless for XAF applications that must work with a legacy database where a file is stored in a BLOB field of the table with other fields of the same object and in a plain form. Also, the same object may contain multiple BLOB fields for different files. 

Reviewer: ```suggestion
The [File Attachments](https://docs.devexpress.com/eXpressAppFramework/112781/document-management/file-attachments-module) module ships with the capability to implement file upload and download functionality in your application. However, the built-in `FileData` class is persistent and therefore requires a separate table in the database to store file contents and names. This means you cannot use this class for XAF applications that work with a legacy database where a file is stored in a BLOB field of the table with other fields of the same object in a plain form - the same object may contain multiple BLOB fields for different files. 
```

Reviewer: I would not recommend that you use the word 'useless' in any context - something can be used or can't be used.

## Implementation Details

> To handle this and other various scenarios, you can create a non-persistent class that implements the `IFileData` interface. This example demonstrates a possible `IFileData` implementation in a non-persistent class that actually passes data to and from the owner persistent object, where the BLOB (`byte[]`) property is declared.

Reviewer: - You do not 'handle' a scenario
- `and other various scenarios` I'm not sure why you use 'various' here
- `that actually passes` What is the difference between "that passes" and "that actually passes"? Does the former mean it does not pass data? 

Rewrite

Reviewer 2: Hi Dirk. Does this variant work?

```suggestion
To execute this and similar scenarios, you can create a non-persistent class that implements the `IFileData` interface. This example demonstrates a possible `IFileData` implementation in a non-persistent class that passes data to/from the owner persistent object where the BLOB (`byte[]`) property is declared.
```

Reviewer: Hi Irina, yes, it looks fine

Important notes:

> - Since the `IFileData` implementor class is not persisent, the object space cannot create its instances. So, we create the only instance in the persistent class constructor.

Reviewer: ```suggestion
- Since the `IFileData` implementor class is not persistent, the object space cannot create its instances, and we create the only instance in the persistent class constructor.
```

- The `IFileData` property must have a setter to allow uploading files in Blazor.

> - In this example, the file name is not stored in the object, but is generated from the object key value. It is only used when the file is downloaded by a user.

Reviewer: ```suggestion
- In this example, the file name is not stored in the object but is generated from the object key value. It is only used when the file is downloaded.
```

> We have created this example only to demonstrate a possible solution. It doesn't cover all possible scenarios. If you need to support an additional scenario, feel free to extend this solution as required (or create your own).

Reviewer: ```suggestion
This example demonstrates only one possible solution. If you need to cover a different scenario, you can extend this solution or create your own.
```
